### PR TITLE
Create framework for varlink endpoint integration tests

### PR DESCRIPTION
**Pull request by baude**
add the ability to write integration tests similar to our e2e tests for
the varlink endpoints.

Signed-off-by: baude <bbaude@redhat.com>

### CODE REVIEW

**Comment by cevich** [comment]
@baude it would be nice to add a one-liner to the docs here about `$SPECIALMODE`: https://github.com/containers/libpod/blob/master/contrib/cirrus/README.md#specialmode

File: contrib/cirrus/setup_environment.sh [hunk 1]
```diff
@@ -73,6 +73,9 @@ case "$SPECIALMODE" in
     none)
         remove_packaged_podman_files
         ;;
+    special)
```
Comment (cevich) [inline contrib/cirrus/setup_environment.sh]: typo: this should be `endpoint`

**Comment by cevich** [comment]
@baude try

```diff
diff --git a/Makefile b/Makefile
index c4b66e26..42dfabe6 100644
--- a/Makefile
+++ b/Makefile
@@ -232,7 +232,7 @@ testunit: libpodimage ## Run unittest on the built image
 localunit: test/goecho/goecho varlink_generate
        ginkgo \
                -r \
-               --skipPackage test/e2e,pkg/apparmor \
+               --skipPackage test/e2e,pkg/apparmor,test/endpoint \
                --cover \
                --covermode atomic \
                --tags "$(BUILDTAGS)" \
```

**Comment by baude** [comment]
please merge

**Comment by TomSweeneyRedHat** [comment]
LGTM and happy green test buttons.  @cevich what sayest thou?

**Pull request merged**